### ISSUE TITLE
Support auto formatting C #

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#57051]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Support-auto-formatting-C/1544648)._

---
Current auto formatting can produce different looking documents depending on their initial state. It would be great to have a formatter that produces unique results for the same logic, no matter how the input is formatted. Users can then provide a set of rules that control how the result should look.

The most helpful feature would be limiting column width (e.g. max 80 columns). Matching the formatting feature from Resharper can be a good first step.

---
#### Feedback Bot on 10/5/2021, 05:34 AM: 

We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.

**Comment by mikadumont** [comment]
Reopening this issue so we can start tracking auto format asks.

**Comment by jmarolf** [comment]
I agree, it feels like this person is really asking for something like https://github.com/belav/csharpier in Visual Studio

**Comment by CyrusNajmabadi** [comment]
We have dotnet format for this now.
